Orange/util.py

```diff
@@ -0,0 +1,68 @@
+"""Various small utilities that might be useful everywhere"""
+
+from functools import wraps
+import sys
+import abc
+import numpy as np
+
+
+def scale(values, min=0, max=1):
+    """Return values scaled to [min, max]"""
+    ptp = np.nanmax(values) - np.nanmin(values)
+    if ptp == 0:
+        return np.clip(values, min, max)
+    return (-np.nanmin(values) + values) / ptp * (max - min) + min
+
+
+def abstract(obj):
+    """Designate decorated class or method abstract."""
+    if isinstance(obj, type):
+        old__new__ = obj.__new__
+
+        def _refuse__new__(cls, *args, **kwargs):
+            if cls == obj:
+                raise NotImplementedError('Can\'t instantiate abstract class ' + obj.__name__)
+            return old__new__(cls, *args, **kwargs)
+
+        obj.__new__ = _refuse__new__
+        return obj
+    else:
+        return abc.abstractmethod(obj)
+
+
+def cached(func):
```

Comment:
Is this decorator really suitable for functions that receive arguments? If I use it to decorate a function sum, and then sum different lists, the result will always be the same (the sum of the first list).

Even when fixed, why is it better than itertools.lru_cache(maxsize=1)?